Userland/Libraries/LibCore/System.cpp

```diff
@@ -1758,6 +1762,14 @@ ErrorOr<String> current_executable_path()
     auto ret = _NSGetExecutablePath(path, &size);
     if (ret != 0)
         return Error::from_errno(ENAMETOOLONG);
+#elif defined(AK_OS_HAIKU)
+    image_info info;
+    for (int32 cookie { 0 }; get_next_image_info(B_CURRENT_TEAM, &cookie, &info) == B_OK && info.type != B_APP_IMAGE;)
+        ;
+    if (info.type != B_APP_IMAGE)
+        return Error::from_string_view("current_executable_path() failed"sv);
+    else
+        memcpy(path, info.name, sizeof(info.name));
```

Comment:
Thanks for the hint,I'm not an expert at C++.
I added an check if info.name is bigger than path and then throw an error,otherwise I'm using strcpy now.